massa-ledger-exports/src/key.rs

```diff
@@ -1,58 +1,77 @@
 use massa_models::{
-    address::{Address, AddressDeserializer, ADDRESS_SIZE_BYTES},
+    address::{Address, AddressDeserializer},
     serialization::{VecU8Deserializer, VecU8Serializer},
 };
-use massa_serialization::{DeserializeError, Deserializer, SerializeError, Serializer};
+use massa_serialization::{Deserializer, SerializeError, Serializer};
 use nom::error::{ContextError, ParseError};
 use std::ops::Bound::Included;
 
 pub const BALANCE_IDENT: u8 = 0u8;
 pub const BYTECODE_IDENT: u8 = 1u8;
 pub const DATASTORE_IDENT: u8 = 2u8;
 
-/// Balance key formatting macro
-#[macro_export]
-macro_rules! balance_key {
-    ($addr:expr) => {
-        [&$addr.prefixed_bytes()[..], &[BALANCE_IDENT]].concat()
-    };
+#[repr(u8)]
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub enum KeyType {
+    BALANCE = 0,
+    BYTECODE = 1,
+    DATASTORE(Option<Vec<u8>>) = 2,
 }
 
-/// Bytecode key formatting macro
-///
-/// NOTE: still handle separate bytecode for now to avoid too many refactor at once
-#[macro_export]
-macro_rules! bytecode_key {
-    ($addr:expr) => {
-        [&$addr.prefixed_bytes()[..], &[BYTECODE_IDENT]].concat()
-    };
+impl KeyType {
+    pub fn from_u8(value: u8) -> Option<Self> {
+        match value {
+            0 => Some(KeyType::BALANCE),
+            1 => Some(KeyType::BYTECODE),
+            2 => Some(KeyType::DATASTORE(None)),
+            _ => None,
+        }
+    }
+
+    pub fn to_u8(&self) -> u8 {
+        match self {
+            KeyType::BALANCE => 0,
+            KeyType::BYTECODE => 1,
+            KeyType::DATASTORE(_) => 2,
+        }
+    }
 }
 
-/// Datastore entry key formatting macro
-///
-/// TODO: add a separator identifier if the need comes to have multiple datastore
-#[macro_export]
-macro_rules! data_key {
-    ($addr:expr, $key:expr) => {
-        [&$addr.prefixed_bytes()[..], &[DATASTORE_IDENT], &$key].concat()
-    };
+/// Disk ledger keys representation
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub struct Key {
```

Comment:
I don't think it's a good idea to rename it from `Key` to `LedgerKey`. We already are in a module related to the Ledger, if we need more context in another module, then we should import `Key` using `ledger_export::Key` to add context. 

We could also import it like so:
```
use ledger_export::Key as LedgerKey;
```
This would allow us to have a clear context. The issue I see with this method is that we will have to use this import in every files using `Key`.

Reply:
Alright so can you use this (use ledger_export::Key as LedgerKey;) in bootstrap (and any other module that are not massa-ledger-XX)?

Reply:
Fixed with https://github.com/massalabs/massa/pull/3455/commits/7b9b651ebb7beb8c425e8f8b0543302744dfae84